src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_CAMERA, outPoly, bgId, pos, NULL, 0x06, 1.0f);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042244.s")
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CD08(CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Actor* actor, Vec3f* pos) {
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_ENTITY, outPoly, bgId, pos, actor, 0x02, 1.0f);
+}
 
-u8 func_80042538(CollisionContext* colCtx, s32 iParm2) {
-    u32 var1 = *(u32*)(iParm2 + 0xc);
-    return var1;
+// Public raycast toward floor
+// returns yIntersect of the poly found, or BGCHECK_Y_MIN if no poly detected
+f32 func_8003CD70(CollisionContext* colCtx, CollisionPoly** outPoly, s32* bgId, Vec3f* pos) {
+    return func_8003C614(NULL, colCtx, COLPOLY_IGNORE_ENTITY, outPoly, bgId, pos, NULL, 0x06, 1.0f);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_bgcheck/func_80042548.s")
+// NTSC 1.0 8002F8E0
+// Tests if moving from `posPrev` to `posNext` will collide with a "wall"
+//`radius` is used to form a sphere for collision detection purposes
+//`checkHeight` is the positive height above posNext to perform certain checks
+// returns true if a collision is detected, else false
+//`outPoly` returns the closest poly detected, while `outBgId` returns the poly owner
+s32 func_8003CDD4(CollisionContext* colCtx, u16 xpFlags, Vec3f* posResult, Vec3f* posNext, Vec3f* posPrev, f32 radius,
+                  CollisionPoly** outPoly, s32* outBgId, Actor* actor, f32 checkHeight, u8 argA) {
+    Lookup* lookupTbl;
+    f32 temp_f0;
+    s32 result;
+    CollisionPoly* poly;
+    f32 dx, dy, dz; // change between posPrev to posNext
+    Vec3f dynaPos;
+    s32 dynaPolyCollision;
+    Vec3f posIntersect;
+    s32 bgId;
+    f32 temp_f0_2;
+    f32 f32temp;
+    f32 nx2, nz2;
+    Vec3f checkLineNext;
+    Vec3f checkLinePrev;
+    f32 n2XZDist;
+    f32 n3XZDist;
+    f32 nx3, nz3;
+    s32 bccFlags;
+    Vec3f posIntersect2;
+    s32 bgId2;
+    f32 nx, ny, nz; // unit normal of polygon
+
+    result = false;
+    *outBgId = BGCHECK_SCENE;
+    *outPoly = NULL;
+    lookupTbl = colCtx->lookupTbl;
+    *posResult = *posNext;
+    dx = posNext->x - posPrev->x;
+    dy = posNext->y - posPrev->y;
+    dz = posNext->z - posPrev->z;
+    if (BgCheck_PosErrorCheck(posNext, "../z_bgcheck.c", 4831) == true ||
+        BgCheck_PosErrorCheck(posPrev, "../z_bgcheck.c", 4832) == true) {
+        if (actor != NULL) {
+            osSyncPrintf("こいつ,pself_actor->name %d\n", actor->id);
+        }
+    }
+    // if we've moved on the xz plane, and ? flag is 0,
+    if ((dx != 0.0f || dz != 0.0f) && (argA & 1) == 0) {
+        if ((checkHeight + dy) < 5.0f) {
+            result = func_8003D7F0(colCtx, xpFlags, COLPOLY_IGNORE_NONE, posPrev, posNext, &posIntersect, &poly, &bgId,
+                                   actor, 1.0f, 0x1B); // bccFlags skip ceiling
```

Comment:
```suggestion
                                   actor, 1.0f, 0x1F & ~BGCHECK_CHECK_CEILING);
```
or or `|` the 4 other flags idk